### PR TITLE
P2 load sample additional

### PR DESCRIPTION
**Pull request by madeye-matt**
Refactored sample service to move async out of SampleServiceImpl and into SampleEndpoint as per earlier PR comment

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java [hunk 1]
```diff
@@ -99,6 +106,29 @@ public SampleEndpoint(SampleService sampleService, MapperFacade mapperFacade) {
 
   }
 
+  /**
+   * Method to kick off a task to ingest a job
+   * @param file Multipart File of SurveySample to be used
+   * @param type Type of Survey to be used
+   * @return a pair containing the newly created SampleSummary and a Future for the long running task
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java]: Doc out of date now

File: src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java [hunk 2]
```diff
@@ -99,6 +106,29 @@ public SampleEndpoint(SampleService sampleService, MapperFacade mapperFacade) {
...
+
+    Callable<Optional<SampleSummary>> callable =() -> {
+      try {
+        return Optional.of(this.sampleService.ingest(newSummary, file, type));
+      } catch (Exception e) {
+        return this.sampleService.failSampleSummary(newSummary, e);
+      }
+    };
+
+    Future<Optional<SampleSummary>> future = EXECUTOR_SERVICE.submit(callable);
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java]: I guess future is unused now

File: src/main/java/uk/gov/ons/ctp/response/sample/service/SampleService.java [hunk 3]
```diff
@@ -79,18 +79,19 @@
...
   /**
-   * Ingest Survey Sample
-   *
+   * Ingest survey sample
+   * @param sampleSummary a newly created samplesummary
    * @param file Multipart File of SurveySample to be used
    * @param type Type of Survey to be used
-   * @throws Exception exception thrown
+   * @return an updated samplesummary
+   * @throws Exception
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/sample/service/SampleService.java]: Might be worth added the scenario it raises an exception in the docs?

File: src/main/java/uk/gov/ons/ctp/response/sample/service/SampleService.java [hunk 4]
```diff
@@ -79,18 +79,19 @@
...
    */
-  Pair<SampleSummary, Future<Optional<SampleSummary>>> ingest(MultipartFile file, String type) throws Exception;
+  SampleSummary ingest(SampleSummary sampleSummary, MultipartFile file, String type) throws Exception;
+
+  Optional<SampleSummary> failSampleSummary(SampleSummary sampleSummary, String message);
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/sample/service/SampleService.java]: Do we still need this method?

File: src/test/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpointUnitTest.java [hunk 5]
```diff
@@ -80,8 +82,12 @@ public void uploadSampleFile() throws Exception {
         // Given
         String type = "B";
         MockMultipartFile file = new MockMultipartFile("file", "filename.txt".getBytes());
+        SampleSummary summary = new SampleSummary();
+        UUID summaryId = UUID.randomUUID();
+        summary.setId(summaryId);
```
Comment (mfcrocker) [inline src/test/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpointUnitTest.java]: There's not much reason to assign the UUID to a variable here, it's not used further down.

Comment (madeye-matt) [reply src/test/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpointUnitTest.java]: Done

File: src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java (hunk 1 above)
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java]: Done

File: src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java (hunk 2 above)
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java]: Done

File: src/main/java/uk/gov/ons/ctp/response/sample/service/SampleService.java (hunk 3 above)
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/sample/service/SampleService.java]: Done

**Pull request merged**